plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/Utils.hs

```diff
@@ -76,9 +79,10 @@ type Arity = [ParamKind]
 data VarInfo tyname name uni fun ann =
   MkVarInfo {
     varStrictness :: Strictness
-    ,varDef       :: Term tyname name uni fun ann
+    , varDef      :: Term tyname name uni fun ann
     -- ^ its definition that has been unconditionally inlined.
-    , arity       :: Arity -- ^ its arity, storing to avoid repeated calculations.
+    , varArity    :: Arity -- ^ its arity, storing to avoid repeated calculations.
+    , varParams   :: [name]
```

Comment:
could consider rolling this into `Arity`: artiy is already a sequence of term/type parameters, including the names in there wouldn't be crazy.